---
title: "Differential Expression Analysis for Ubi-Enriched Peptides."
author: "Functional Genomics Center Zurich"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    toc: true
  bookdown::html_document2:
    toc: true
  pdf_document: 
    toc: true
header-includes: 
  \usepackage{fancyhdr}
  \pagestyle{fancy}
  \fancyhead[CO,CE]{Group Comparison}
  \fancyfoot[CO,CE]{\textbf{FGCZ} - www.fgcz.ch - 2018}
  \fancyfoot[LE,RO]{\thepage} 
params:
  grp: NULL
vignette: >
  %\VignetteIndexEntry{FGCZ Two-Group Analysis}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
bibliography: bibliography.bib
editor_options: 
  chunk_output_type: console
always_allow_html: true
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(
  echo = FALSE,
  message = FALSE,
  warning = FALSE
)



grp2 <- params$grp
```


# B-fabric related information

This report is stored in the LIMS system [_bfabric_](https://fgcz-bfabric.uzh.ch) [@Trker2010] in
__project__: `r grp2$Bfabric$projectID`, __order__: `r grp2$Bfabric$orderID`, `r if(!is.null(grp2$Bfabric$workunitID)){ "with the workunit name :" }`  `r grp2$Bfabric$workunitID`.


The ubi-enriched peptide identification and quantification were performed using: 
`r if(!is.null(grp2$Software)){grp2$Software} else {"<Specify parameter grp2$Software>"}`. The input file can be downloaded from here : [`r grp2$Bfabric$inputID`](`r grp2$Bfabric$inputURL`). The set of parameters used to run the quantification software can be retrieved from the b-fabric workunit to which this file belongs.



# Introduction

The differential expression analysis verifies if the __difference__ between normalized empirical peptide abundances measured in two groups is significantly non-zero.
To make the test as sensitive and specific as possible, the methods used to measure [@taverna2021critical] and estimate peptide abundances [@grossmann2010implementation] are optimized to minimize the biochemical and technical variance. In addition these empirical abundances $A$ are further $log_2$ transformed and scaled to make them compatible with the statistical test procedure [@valikangas2018systematic]. Therefore, we obtain a scale free $\log_2$ transformed normalized peptide abundance $\log_2(A)$ for a sample.

For unpaired experiments the difference $\Delta$ for a specific peptide is estimated by:  

$$
\Delta = \frac{1}{n}\sum_{i=1}^n \log_2(A^a_i) - \frac{1}{m}\sum_{j=1}^{m}\log_2(A^b_j)
$$



where $A^a_i$ - is the normalized peptide abundance of sample $i$ in the group $a$ of $n$ samples, while $A^b_j$ is the peptide abundance of sample $j$ in group $b$ of $m$ samples.

For paired experiment the difference is estimated by:


$$
\Delta = \frac{1}{n}\sum_{i=1}^n \log_2(A^a_i) - \log_2(A^b_{i})
$$


where $n$ is the number of subjects, each treated with $a$ and $b$.

Of note, when comparing two samples $a$ and $b$ the difference of logarithms equals to the logarithm of the ratio (exponent rule):

$$
\log(A^a) - \log(A^b) = \log(\frac{A^a}{A^b})
$$


, it is called $\log$-ratio or $\log$ fold-change (logFC).

The estimated differences $\Delta$ have an associated error $\epsilon$. Therefore, the differential expression analysis must test if the difference is significantly nonzero. 

We run a set of functions implemented in the R package _[prolfqua]_ [@prolfquawolski2023] to filter and normalize the data, generate visualizations, and to compute differential expression analysis. To future improve the power of the differential expression test the peptide variances are moderated [@Smyth2004], i.e. the individual peptide variances are updated using a variance prior estimated from all the peptides in the experiment.

# Results

Table \@ref(tab:samples) shows the number of samples assigned to each group while Table \@ref(tab:annotation) shows the names of the files assigned to the group.

```{r samples, eval=TRUE}

tab <- data.frame(table(grp2$RES$lfqData$factors()[grp2$RES$lfqData$config$table$factor_keys_depth()]))
colnames(tab)[length(colnames(tab))] <- "# samples"
knitr::kable(tab, caption = "Nr of samples assigned to each group.")

```


```{r annotation, eval=TRUE}
knitr::kable(grp2$RES$lfqData$factors(), caption = "LC-MS samples annotation table. The content of the sampleName column is used as a short form to plot labels. The group to which a sample is assigned to is shown in the column group.")

```

## Ubi-enriched Peptide identification

Here Jonas will write more about the input file that is site centric!

The peptide matrix is filtered using the following steps:
- Only di-Gly peptides are kept
- Best Localization Probability: `r grp2$pop$LocProbThresh`
- FGCZContaminant proteins (added to the source fasta database) are filtered out

The overall number of identified proteins with at least one ubi-sites detected in this experiment is: `r grp2$RES$Summary$NrOfProteinsNoDecoys`.
This number can only be estimated if a decoy database is searched, decoys are reported and the decoy pattern is correctly set in the search engine. The decoy pattern is set to: `r grp2$pop$processing_options$pattern_decoys`.

The percentage of contaminant peptides is : `r grp2$RES$Summary$percentOfContaminants` %. The percentage of false positive identifications (Decoy sequences) is `r grp2$RES$Summary$percentOfFalsePositives` %.

To identify a high number of ubi-peptides an enrichment strategy was used. The ratio of ubi-peptides versus non-di-Gly peptides identified was:  __`r round(enrichment_ratio, 2)` percent__.


We keep the contaminant peptides because, for some experiments, these contaminants are relevant. However, they can be recognized since their identifiers start with __zz__ or __CON__.
We also keep the decoy sequences because they allow us to re-estimate the proportion of falsely identified peptides in the list of differentially expressed peptides, which might differ from that of entire dataset. The identifiers of the decoy peptides start with __REV__ or __rev__.


Figure \@ref(fig:nrPerSample) shows the number of quantified peptides per sample.

```{r nrPerSample, fig.cap="Number of identified peptides across samples.", fig.with=10, fig.height=7}
sum <- grp2$RES$lfqData$get_Summariser()
sum$plot_hierarchy_counts_sample()

```

## Missing Value Analysis

The absence of a peptide measurement in a sample might be biologically relevant or might point to technical problems. Significant differences in the set of peptides observed in the samples within a group typically indicate either technical problems or excessive biological variability. If one sample out of ten has a different set of peptides, it is likely an outlier and can be removed from the analysis. If the differences between the groups are significant but within the groups are small, this might systematically bias the difference estimates, i.e., produce false-positive or false-negative test results.


```{r countProtWithNAs}
res <- grp2$RES$lfqData$to_wide(as.matrix = TRUE)$data
res[!is.na(res)] <- 0
res[is.na(res)] <- 1
allrows <- nrow(res)
res <- res[apply(res,1, sum) > 0, , drop = FALSE]
allNas <- nrow(res)
```


```{r prepHeatmap}
pl <- grp2$RES$lfqData$get_Plotter()

nah <- pl$NA_heatmap()


```

A dichotomous view of the data can be constructed by transforming peptide abundance estimates into present/absent calls (Figure \@ref(fig:naHeat) ). The heatmap shows only peptides with at least one missing value. There are `r allNas` peptides with at least one missing value in the data, which is (`r allNas/allrows * 100` %).

We expect that samples in the same group are more similar and cluster together, i.e., they are in the same branch of the dendrogram.

(ref:naHeat) peptide abundance heatmap (rows indicate peptides, columns indicate samples) showing missing peptide abundance estimates across data set. Rows and columns are grouped based on the Minkowski distance using hierarchical clustering. White: peptide is observed, black: peptide is not observed.

```{r naHeat, fig.width=7, fig.height=6, dpi=300, fig.cap="(ref:naHeat)", eval=TRUE}
nah
```

Using Figure \@ref(fig:vennPeptides) we examine if we see the same ubi-enriched peptide in each group. We say a ubi-enriched peptide is unobserved in the group if it is absent in all samples and is present otherwise. A significant overlap among groups allows more precisely estimating the peptide abundance differences between the groups.

(ref:vennPeptides) Venn diagram showing the number of peptides present in each group and in all possible intersections among groups. 

```{r vennPeptides, fig.cap="(ref:vennPeptides)"}
pups <- prolfqua::UpSet_interaction_missing_stats(grp2$RES$lfqData$data, grp2$RES$lfqData$config, tr = 1)
UpSetR::upset(pups$data, order.by = "freq", nsets = pups$nsets)

```


## Peptide Abundance Analysis

The density plot (Figure \@ref(fig:normalized) left panel)  displays the peptide abundance distribution for all data set samples. Mayor differences between samples could hint that the individual peptide abundance values are affected by technical biases. These biases might need to be corrected to separate them from biological effects. The right panel of Figure \@ref(fig:normalized) shows the distribution of the transformed and scaled normalized empirical peptide abundances. Normalization is applied to remove systematic differences in peptide abundances due to different sample concentrations or amounts of sample loaded on a column. However, in the presence of a large proportion of missing data, normalization potentially amplifies systematic errors.


```{r normalization, results = 'asis'}
if (grp2$processing_options$transform == "robscale") {
  cat("To do this the z-score of the $\\log_2$ transformed peptide abundances are computed.",
      "Because we need to estimate the peptide differences on the original scale, we have to multiply the $z$-score by the average standard deviation of all the $N$ samples in the experiment.",
  "After normalization all samples have an equal mean and variance and a similar distribution.")
} else if (grp2$processing_options$transform == "vsn") {
  cat("To do this the variance stabilizing normalization (vsn) was applied [@HuberVSN2002].")
} else if (grp2$processing_options$transform == "none") {
  cat("However, in some circumstances it is advisable not to normalize the data, e.g. in case of affinity purification experiments, or when the requirements of sufficient similarity among samples are not met.")
}

```

(ref:normalized) Kernel density function showing the distribution of peptide abundances in all samples. Left panel : Empirical peptide abundance of all samples in the dataset. Right panel: Normalized empirical peptide abundances of all samples in the dataset.

```{r normalized,  fig.width=8, fig.height=5,dpi=300, fig.cap="(ref:normalized)", eval=TRUE}

pl <- grp2$RES$lfqData$get_Plotter()
p1 <- pl$intensity_distribution_density()

plTransformed <- grp2$RES$transformedlfqData$get_Plotter()
p2 <- plTransformed$intensity_distribution_density() +
  ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 90, vjust = 0.5, hjust = 1))

gridExtra::grid.arrange(p1, p2, nrow = 1)

```

The median coefficient of variation (CV) of a group of samples or all samples in the dataset, can be used to compare the experiment with other experiments [@piehowski2013sources]. For example, the median CV for high-performance liquid chromatography experiments ranges from 2% to 35% depending on the biological samples studied, the chromatography method used, label-free or labelled quantification [@taverna2021critical].


Figure \@ref(fig:SDViolin), show the coefficients of variations (CV) for all peptides computed on not normalized data. Ideally the within group CV should be smaller than the CV of all samples. 

(ref:SDViolin) Distribution of coefficient of variation (CV) within each groups and in the entire experiment (all).

```{r SDViolin, fig.height=3, fig.width=7, fig.cap="(ref:SDViolin)", eval=TRUE}
stR <- grp2$RES$lfqData$get_Stats()
pA <- stR$violin() + ggplot2::theme_classic() + ggplot2::labs(tag = "A")
pA

```

The Table \@ref(tab:CVtable) shows the median CV of all groups and across all samples (all).

```{r CVtable, eval=TRUE}
resR <- stR$stats_quantiles()
C <- dplyr::bind_rows(CV = resR$wide  |> dplyr::filter(probs == 0.5) |> round(digits = 2))
C <- C |> dplyr::mutate( what = c("CV"), .before = 1 )
C$probs <- NULL
knitr::kable(C, caption = 'Median of coefficient of variation (CV).')

```



```{r generateHeatmaps, inlcude = FALSE}
pl <- grp2$RES$transformedlfqData$get_Plotter()
ph <- pl$heatmap()

```

The peptide abundance heatmap (Figure \@ref(fig:heatmap)) groups the peptide and samples using unsupervised hierarchical clustering. Distances between peptides and samples are computed using normalized peptide abundances. Peptides with a large proportion of missing observations are not shown in this heatmap, because for these peptides no distance can be computed.  Peptides and samples showing similar abundances are grouped and shown in adjacent rows and columns respectively. 


(ref:heatmap) Peptide abundance heatmap (rows indicate peptides, columns indicate samples) showing the row scaled $\log_2$ transformed peptide abundance value. Co-clustering (hierarchical complete linkage, euclidean distance) of samples and peptides was used.



```{r heatmap, fig.width=7, fig.height=8, dpi=300, fig.cap="(ref:heatmap)", eval=TRUE}
ph
```

We use principal component analysis (PCA) to transform the high dimensional space defined by all peptides into a two-dimensional one containing most of the information. Plot \@ref(fig:pca) shows the location of the samples according to the first and second principal component, which explain most of the variance in the data. Samples close in the PCA plot are more similar than those farther apart.

(ref:pca)  Plot of first and second principal component (PC1 and PC2) of principal component analysis (PCA). Normalized abundances were used as input.

```{r pca, fig.cap = "(ref:pca)" , fig.width=7, fig.height=7 }
grp2$RES$transformedlfqData$get_Plotter()$pca_plotly()
```

# Differential Expression Analysis

The method used to test for differential expression consists of several steps: First a linear model that explains the observed peptide abundances using the grouping of the samples is fitted using the R function _lm_ to each peptide:

`r grp2$RES$formula`.

Secondly, the difference between the group is computed (Table \@ref(tab:contrtable)).

```{r contrtable}
df <- data.frame(name = names(grp2$pop$Contrast), contrast = grp2$pop$Contrast)
rownames(df) <- NULL
knitr::kable(df, caption = "Name of difference (Contrast), and formula used to compute it.")

```


and a null hypothesis significance test (NHST) is conducted, where the null hypothesis is that the peptide is not differentially expressed [@faraway2004linear]. 

If there are no abundances measured in one of the groups for some peptides, we assume the observations are missing because the peptide abundance is below the detection limit. Therefore, we estimate the detection limit using the mean of the $1\%$ smallest group averages. 
Furthermore, to make it explicit for which peptides we did impute the unobserved group mean, we label them with `Imputed_Mean` (see table in Figures \@ref(fig:tableAllProt) column `modelName`) and visualize them with gray dots in Figure \@ref(fig:volcanoplot). Finally, those peptides with a sufficiently large number of observations are labeled with `Linear_Model_Moderated`.

Next, to increase the power of the analysis variance shrinkage is performed [@Smyth2004]. Finally, the false discovery rate (FDR) using the Benjamini-Hochberg procedure is computed [@benjamini1995controlling].
The FDR is the expected proportion of false discoveries in a list of peptides, and can be used to select candidates for follow up experiments. FDR thresholds commonly used are 5, 10 or 25%. By filtering the peptides using a FDR threshold of `r grp2$pop$FDRthreshold*100` % we can expect this proportion of false positives in the list and `r (1 - grp2$pop$FDRthreshold) * 100` % truly differentially expressed peptides. Because we do not know which of them are true positives follow up experiments are necessary.

The table (Figure \@ref(fig:tableAllProt)) summarizes the differential expression analysis results by providing the following information:

- protein_Id - protein identifier from database search
- site - the PTM-site identifier in the protein
- description - information about the protein provided in the FASTA database
- contrast - name of the comparison
- modelName - name of the method to estimate differences : Imputed_mean or Linear_Model_Moderated
- FDR - false discovery rate
- diff - difference between groups.

The volcano plot \@ref(fig:volcanoplot) helps to identify peptides with a large differences among groups and a low FDR. The significance dimension is a $-\log_{10}$ transformed FDR, i.e., small values of FDR become large after transformation. Promising candidate peptides are found in the upper right and left sector of the plot.


```{r setupVolcano, echo = FALSE}
datax <- as.data.frame(grp2$RES$contrastsData)
```


```{r tableAllProt, fig.cap="Differential expression analysis results of all peptides."}
bb <- datax |> dplyr::select(
  all_of(c("site",
           "protein_Id",
           "description",
           "contrast",
           "modelName",
           "diff",
           "FDR" )))
bb <- crosstalk::SharedData$new(bb, ~site, group = "BB")

DT::datatable(bb, filter = "bottom",
  extensions = "Scroller",
  style = "auto",
  class = "compact",
  options = list(deferRender = TRUE,
                 scrollY = 300,
                 scrollX = 400,
                 scroller = TRUE) ) |>
  DT::formatRound(columns = c('FDR', 'diff'), digits = 3)

```

(ref:volcanoplot)  Volcano plot showing $-\log_{10}$ transformed FDR as function of the difference between groups. The red line indicates the $-log_{10}(FDR)$ of FDR = `r grp2$pop$FDRthreshold`, while the green lines represent the the difference of minus and plus `r grp2$pop$Diffthreshold`. With orange dots differences and FDRs estimated using missing value imputation are shown.


```{r volcanoplot, fig.cap = "(ref:volcanoplot)", fig.width=9, fig.height=7, include = TRUE, eval = TRUE}

palette <- c("black","orange")
palette <- setNames(palette, c("Linear_Model_moderated", "Imputed_Mean_moderated"))  
xd <- prolfqua::volcano_plotly( 
  datax ,
  proteinID = "site",
  effect = "diff",
  significance = "FDR",
  contrast = "contrast",
  color = "modelName",
  palette = palette,
  xintercept =  c(-grp2$processing_options$diff_threshold,grp2$processing_options$diff_threshold),
  yintercept = grp2$processing_options$FDR_threshold,
  title_size = 10,
  group = "BB")

xd <- lapply(xd, plotly::highlight , off = "plotly_doubleclick")
nrow <- ceiling(length(xd) / 4)
plotly::subplot(xd, shareX = TRUE, shareY = TRUE, nrows = nrow)

```

## Differentially Expressed Ubi-Sites

Here we use the FDR threshold of `r grp2$processing_options$FDR_threshold` and a difference threshold of `r grp2$processing_options$FDR_threshold` to select differentially expressed peptides. Table \@ref(tab:nrsignificant) summarizes the number of significant calls.

```{r nrsignificant, results="markup", eval=TRUE}
library(dplyr)
mx <- datax |> dplyr::mutate(passes = abs(diff) >  grp2$processing_options$FDR_threshold  & FDR < grp2$processing_options$FDR_threshold)
x <- mx |> dplyr::group_by(contrast) |> summarize(n = n(), Significant = sum(passes), "Not Significant" = n() - sum(passes))

mycap <- paste0("Number of not significant and significant ubi-enriched sites."  )
knitr::kable(x, caption = mycap)

```



```{r areThereSig}
datax_signif <- grp2$RES$contrastsData_signif
showSignificant <- TRUE
if (nrow(datax_signif) == 0) {
  showSignificant <- FALSE
}

```


```{r results = 'asis', eval = showSignificant}
cat("The table shown in Figure \\@ref(fig:SigPrey) lists all the significant sites")
```

(ref:SigPrey) Significant ubi-enriched sites obtained by applying the difference and FDR thresholds.


```{r SigPrey, fig.cap= "(ref:SigPrey)", eval = showSignificant}
ctdata <- datax_signif |> dplyr::select(all_of(c("site","protein_Id" ,  "description", "contrast", "modelName", "diff", "FDR"  ))) |> as.data.frame()

sig <- crosstalk::SharedData$new(ctdata, ~site, group = "BB")
DT::datatable(sig, filter = "bottom",
  extensions = "Scroller",
  style = "auto",
  class = "compact",
  options = list(deferRender = TRUE,
                 scrollY = 300,
                 scrollX = 400,
                 scroller = TRUE) ) |> 
  DT::formatRound(columns = c('FDR', 'diff'), digits = 3)
```




```{r prepareHeatmap, eval = showSignificant}
signif <- grp2$RES$transformedlfqData$get_copy()
signif <- signif$get_subset(datax_signif)
if (signif$hierarchy_counts()[2] > 30) {
  rownames = FALSE
} else {
  rownames = TRUE
}
sigheat <- signif$get_Plotter()$raster(rownames = rownames)

```


```{r makeText, eval = showSignificant, results = 'asis'}

cat(paste("Furthermore, Figure \\@ref(fig:sigpeptides) shows a heatmap of log2 transformed peptide abundances of all significant calls."))

```

(ref:sigpeptides) Heatmap showing the $\log_2$ transformed peptide abundances for peptides which pass the FDR and difference thresholds.

```{r sigpeptides, fig.cap="(ref:sigpeptides)", eval = showSignificant}
sigheat
```

(ref:vennDiagramSig) Venn diagram showing the number of significant ubi-enriched sites for each contrast and their intersections.


```{r vennDiagramSig, fig.cap="(ref:vennDiagramSig)", eval = showSignificant}
xx <- split(datax_signif$site, datax_signif$contrast)
if (length(xx) > 1) {
  UpSetR::upset(UpSetR::fromList(xx))
}

```


# Additional Analysis

The zip file contains an excel file __Results.xlsx__. All the figures can be recreated using the data in the excel file. The Excel file contains the following spreadsheets:

- __annotation__ - the annotation of the samples in the experiment
- __raw_abundances__ table with empirical ubi-enriched site abundances.
- __normalized_abundances__ table with normalized ubi-enriched site abundances. 
- __raw_abundances_matrix__ A table where each column represents a sample and each row represents a ubi-enriched site and the cells store the empirical ubi-enriched site abundances.
- __normalized_abundances_matrix__ A table where each column represents a sample and each row represents a ubi-enriched site and the cells store the empirical ubi-enriched site abundances.
- __diff_exp_analysis__ A table with the results of the differential expression analysis. For each ubi-enriched site there is an row containing the estimated difference between the groups, the false discovery rate FDR, the 95% confidence interval, the posterior degrees of freedom. 
- __missing_information__ - spreadsheet containing information if a ubi-enriched site is present (1) or absent in a group (0).
- __ubi-enriched site_variances__ - spreadsheet which for each ubi-enriched site shows the variance (var) or standard deviation (sd) within a group, the number of samples (n) and the number of observations (not_na) as well as the group average intensity (mean).


The zip file also can includes an "boxplot.pdf" file which shows the boxplots of normalized ubi-enriched site abundances for all the ubi-sites in the dataset.

For questions and improvement suggestions, with respect to this report, please do contact <protinf@fgcz.uzh.ch>.


# Session Information

```{r sessionInfo}
pander::pander(sessionInfo())
```


# References {-}

<div id="refs"></div>


# Glossary

- groups - different treatments, genotypes etc.
- diff (difference) - it is the difference of the ubi-site-peptide abundance estimate of two groups
- FDR - false discovery rate




