<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ prophosqua</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">prophosqua</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/Supplementary_Material_v3.html">Integrated Analysis of Post-Translational Modifications and Total Proteome: Methods for Distinguishing Abundance from Usage Changes</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">More articles...</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/prolfqua/prophosqua/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>



      <dl></dl></div><div class="section level2">




      <dl><dt>

          <code><a href="copy_phospho_integration.html">copy_phospho_integration()</a></code>

        </dt>
        <dd>copy Markdown and runscript for FragPipe combined_protein.tsv</dd>

        <dt>

          <code><a href="dea_res_dir.html">dea_res_dir()</a></code>

        </dt>
        <dd>get xlsx path</dd>

        <dt>

          <code><a href="dea_xlsx_path.html">dea_xlsx_path()</a></code>

        </dt>
        <dd>get xlsx path</dd>

        <dt>

          <code><a href="derive_motifs_from_gmt.html">derive_motifs_from_gmt()</a></code>

        </dt>
        <dd>Derive motifs from PTMsigDB GMT file</dd>

        <dt>

          <code><a href="dot-test_diff_diff.html">.test_diff_diff()</a></code>

        </dt>
        <dd>Test if differences of differences are significant</dd>

        <dt>

          <code><a href="download_ptmsigdb.html">download_ptmsigdb()</a></code>

        </dt>
        <dd>Download PTMsigDB signatures</dd>

        <dt>

          <code><a href="exampleN_C_dat.html">exampleN_C_dat</a></code>

        </dt>
        <dd>Example dataset exampleN_C_dat</dd>

        <dt>

          <code><a href="exampleN_C_dat_no_prot.html">exampleN_C_dat_no_prot</a></code>

        </dt>
        <dd>Example dataset exampleN_C_dat_no_prot</dd>

        <dt>

          <code><a href="explode_multisites.html">explode_multisites()</a></code>

        </dt>
        <dd>each site goes into own row, if more then on site per multisite</dd>

        <dt>

          <code><a href="filter_contaminants.html">filter_contaminants()</a></code>

        </dt>
        <dd>filter contaminants</dd>

        <dt>

          <code><a href="get_kinase_motifs.html">get_kinase_motifs()</a></code>

        </dt>
        <dd>Get default list of kinase motifs</dd>

        <dt>

          <code><a href="get_sequence_windows.html">get_sequence_windows()</a></code>

        </dt>
        <dd>extract sequence windows, from sequence at pos in protein</dd>

        <dt>

          <code><a href="load_and_preprocess_data.html">load_and_preprocess_data()</a></code>

        </dt>
        <dd>load_and_preprocess_data</dd>

        <dt>

          <code><a href="make_fasta_summary.html">make_fasta_summary()</a></code>

        </dt>
        <dd>Make FASTA database summary</dd>

        <dt>

          <code><a href="n_c_integrated_df.html">n_c_integrated_df</a></code>

        </dt>
        <dd>Example dataset n_c_integrated_df</dd>

        <dt>

          <code><a href="n_to_c_expression.html">n_to_c_expression()</a></code>

        </dt>
        <dd>Plot protein and PTM expression</dd>

        <dt>

          <code><a href="n_to_c_expression_multicontrast.html">n_to_c_expression_multicontrast()</a></code>

        </dt>
        <dd>Plot protein and PTM expression across all contrasts in multi-panel layout</dd>

        <dt>

          <code><a href="n_to_c_plot.html">n_to_c_plot()</a></code>

        </dt>
        <dd>N to C plot using ggplot2</dd>

        <dt>

          <code><a href="n_to_c_plot_integrated.html">n_to_c_plot_integrated()</a></code>

        </dt>
        <dd>N to C for integrated results</dd>

        <dt>

          <code><a href="n_to_c_usage.html">n_to_c_usage()</a></code>

        </dt>
        <dd>Plot PTM occupation, use after adjusting for total proteome</dd>

        <dt>

          <code><a href="n_to_c_usage_multicontrast.html">n_to_c_usage_multicontrast()</a></code>

        </dt>
        <dd>Plot PTM usage across all contrasts in multi-panel layout</dd>

        <dt>

          <code><a href="plot_diff_logo.html">plot_diff_logo()</a></code>

        </dt>
        <dd>Plot Difference Sequence Logo (Upregulated - Downregulated)</dd>

        <dt>

          <code><a href="plot_seqlogo_with_diff.html">plot_seqlogo_with_diff()</a></code>

        </dt>
        <dd>Plot Sequence Logos with Difference Column</dd>

        <dt>

          <code><a href="ptmsea_data_prep.html">ptmsea_data_prep()</a></code>

        </dt>
        <dd>Prepare data for PTM-SEA analysis</dd>

        <dt>

          <code><a href="ptmsea_ora_prep.html">ptmsea_ora_prep()</a></code>

        </dt>
        <dd>Prepare data for PTM-SEA ORA analysis</dd>

        <dt>

          <code><a href="ptmsigdb_to_term2gene.html">ptmsigdb_to_term2gene()</a></code>

        </dt>
        <dd>Convert PTMsigDB pathways to TERM2GENE format</dd>

        <dt>

          <code><a href="read_kinase_motifs_csv.html">read_kinase_motifs_csv()</a></code>

        </dt>
        <dd>Read kinase motifs from CSV</dd>

        <dt>

          <code><a href="run_motif_gsea.html">run_motif_gsea()</a></code>

        </dt>
        <dd>Run Motif-based GSEA</dd>

        <dt>

          <code><a href="run_ptmsea.html">run_ptmsea()</a></code>

        </dt>
        <dd>Run PTM-SEA without direction</dd>

        <dt>

          <code><a href="run_ptmsea_up_down.html">run_ptmsea_up_down()</a></code>

        </dt>
        <dd>Run directional PTM-SEA analysis</dd>

        <dt>

          <code><a href="scan_motifs.html">scan_motifs()</a></code>

        </dt>
        <dd>Scan sequences for kinase motifs</dd>

        <dt>

          <code><a href="split_ptmsigdb_pathways.html">split_ptmsigdb_pathways()</a></code>

        </dt>
        <dd>Run PTM-SEA analysis</dd>

        <dt>

          <code><a href="test_diff.html">test_diff()</a></code>

        </dt>
        <dd>compute MSstats like test statsitics</dd>

        <dt>

          <code><a href="trim_ptmsigdb_pathways.html">trim_ptmsigdb_pathways()</a></code>

        </dt>
        <dd>Trim PTMsigDB pathways to shorter flanking sequences</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jonas Grossmann, Witold Wolski.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer></div>





  </body></html>

