<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Motif-Based Enrichment Analysis: Kinase Activity Inference • prophosqua</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Motif-Based Enrichment Analysis: Kinase Activity Inference">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">prophosqua</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Supplementary_Material_v3.html">Integrated Analysis of Post-Translational Modifications and Total Proteome: Methods for Distinguishing Abundance from Usage Changes</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">More articles...</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Motif-Based Enrichment Analysis: Kinase Activity Inference</h1>
            
      

      <div class="d-none name"><code>Analysis_Motif_Enrichment.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>While PTM-SEA relies on curated databases like PTMsigDB, many
phosphosites do not map to known substrates. Motif-based analysis
provides a complementary approach by scanning the <em>sequence
context</em> of observed phosphosites for known kinase consensus motifs
(e.g., matching the <code>R-x-x-S/T</code> pattern for Basophilic
kinases).</p>
<p>This vignette demonstrates how to: 1. Load a database of kinase regex
motifs. 2. Scan your phosphoproteomics sequence windows for matches. 3.
Perform GSEA to identify enriched kinase motifs in your contrasts.</p>
</div>
<div class="section level2">
<h2 id="load-data">Load Data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">prophosqua</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/contribution-knowledge-mining/" class="external-link">clusterProfiler</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/contribution-knowledge-mining/" class="external-link">enrichplot</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ctlab/fgsea/" class="external-link">fgsea</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://purrr.tidyverse.org/" class="external-link">purrr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load example data (same as PTM-SEA vignette)</span></span>
<span><span class="va">example_path</span> <span class="op">&lt;-</span> <span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org/reference/here.html" class="external-link">here</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"combined_test_diff_example.rds"</span><span class="op">)</span></span>
<span><span class="va">example_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">example_path</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span></span>
<span>  Property <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Rows"</span>, <span class="st">"Columns"</span>, <span class="st">"Contrasts"</span><span class="op">)</span>,</span>
<span>  Value <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">example_data</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">example_data</span><span class="op">)</span>,</span>
<span>            <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">example_data</span><span class="op">$</span><span class="va">contrast</span><span class="op">)</span>, collapse <span class="op">=</span> <span class="st">", "</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">data_info</span>, caption <span class="op">=</span> <span class="st">"Example Data"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Example Data</caption>
<colgroup>
<col width="12%">
<col width="87%">
</colgroup>
<thead><tr class="header">
<th align="left">Property</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Rows</td>
<td align="left">105824</td>
</tr>
<tr class="even">
<td align="left">Columns</td>
<td align="left">56</td>
</tr>
<tr class="odd">
<td align="left">Contrasts</td>
<td align="left">KO_vs_WT, KO_vs_WT_at_Early, KO_vs_WT_at_Late,
KO_vs_WT_at_Uninfect</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="load-and-define-motifs">Load and Define Motifs<a class="anchor" aria-label="anchor" href="#load-and-define-motifs"></a>
</h2>
<p>We use <code><a href="../reference/get_kinase_motifs.html">get_kinase_motifs()</a></code> to retrieve a default set of
consensus patterns (e.g., from literature/Scansite/ELM). You can also
add your own.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load kinase motifs (internal database + optional user extensions)</span></span>
<span><span class="va">motif_db</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_kinase_motifs.html">get_kinase_motifs</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Example of adding a custom motif</span></span>
<span><span class="va">my_motif</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  kinase_group <span class="op">=</span> <span class="st">"Custom"</span>,</span>
<span>  motif_name <span class="op">=</span> <span class="st">"MySpecialKinase"</span>,</span>
<span>  pattern <span class="op">=</span> <span class="st">"R..S..[DE]"</span>,</span>
<span>  description <span class="op">=</span> <span class="st">"Custom motif"</span>,</span>
<span>  stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">motif_db</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_kinase_motifs.html">get_kinase_motifs</a></span><span class="op">(</span>extra_motifs <span class="op">=</span> <span class="va">my_motif</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">motif_db</span><span class="op">)</span>, caption <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Kinase Motifs ("</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">motif_db</span><span class="op">)</span>, <span class="st">"total)"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Kinase Motifs ( 17 total)</caption>
<colgroup>
<col width="13%">
<col width="18%">
<col width="17%">
<col width="51%">
</colgroup>
<thead><tr class="header">
<th align="left">kinase_group</th>
<th align="left">motif_name</th>
<th align="left">pattern</th>
<th align="left">description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Basophilic</td>
<td align="left">PKA_AKT_consensus</td>
<td align="left">R.R..[ST]…..</td>
<td align="left">R-x-R-x-x-S/T (PKA, AKT, etc. - strong basophilic)</td>
</tr>
<tr class="even">
<td align="left">Basophilic</td>
<td align="left">PKA_Classic</td>
<td align="left">R.[ST]..</td>
<td align="left">R-x-S/T (PKA classic)</td>
</tr>
<tr class="odd">
<td align="left">Basophilic</td>
<td align="left">PKC_General</td>
<td align="left">..[ST].[RK]</td>
<td align="left">S/T-x-R/K (PKC general)</td>
</tr>
<tr class="even">
<td align="left">Basophilic</td>
<td align="left">Akt_RRxS</td>
<td align="left">RR.[ST]…</td>
<td align="left">R-R-x-S/T (Akt preference)</td>
</tr>
<tr class="odd">
<td align="left">Basophilic</td>
<td align="left">CAMK2</td>
<td align="left">R..[ST]…</td>
<td align="left">R-x-x-S/T (CaMKII)</td>
</tr>
<tr class="even">
<td align="left">Basophilic</td>
<td align="left">Substrate_RK_rich</td>
<td align="left">[RK][RK].[ST]…</td>
<td align="left">Basic residues upstream (R/K-R/K-x-S/T)</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="motif-scanning">Motif Scanning<a class="anchor" aria-label="anchor" href="#motif-scanning"></a>
</h2>
<p>Instead of relying on exact PTMsigDB matches (which often have low
overlap), we scan our <em>observed</em> sequence windows for these
kinase motifs.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get unique sequences from our data</span></span>
<span><span class="co"># Ensure we use 15-mers if possible, or whatever length is appropriate.</span></span>
<span><span class="co"># Default scanning assumes the central residue (pos 8) is the phosphosite.</span></span>
<span><span class="va">our_sequences</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">toupper</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/trimws.html" class="external-link">trimws</a></span><span class="op">(</span><span class="va">example_data</span><span class="op">$</span><span class="va">SequenceWindow</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Scan for motifs</span></span>
<span><span class="co"># center_pos = 8 ensures we only count matches where the motif overlaps the actual phosphosite</span></span>
<span><span class="va">motif_matches</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scan_motifs.html">scan_motifs</a></span><span class="op">(</span><span class="va">our_sequences</span>, motif_db <span class="op">=</span> <span class="va">motif_db</span>, center_pos <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span>
<span></span>
<span><span class="va">scan_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span></span>
<span>  Metric <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Total matches"</span>, <span class="st">"Unique sites"</span><span class="op">)</span>,</span>
<span>  Value <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">motif_matches</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">motif_matches</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">scan_info</span>, caption <span class="op">=</span> <span class="st">"Scanning Results"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Scanning Results</caption>
<thead><tr class="header">
<th align="left">Metric</th>
<th align="right">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Total matches</td>
<td align="right">30454</td>
</tr>
<tr class="even">
<td align="left">Unique sites</td>
<td align="right">16761</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Top motifs</span></span>
<span><span class="va">top_motifs</span> <span class="op">&lt;-</span> <span class="va">motif_matches</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">term</span>, name <span class="op">=</span> <span class="st">"Count"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">Count</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>Motif <span class="op">=</span> <span class="va">term</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">top_motifs</span>, caption <span class="op">=</span> <span class="st">"Top 10 Motifs by Match Count"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Top 10 Motifs by Match Count</caption>
<thead><tr class="header">
<th align="left">Motif</th>
<th align="right">Count</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Proline_General</td>
<td align="right">6762</td>
</tr>
<tr class="even">
<td align="left">GSK3_Primed</td>
<td align="right">3370</td>
</tr>
<tr class="odd">
<td align="left">CAMK2</td>
<td align="right">3250</td>
</tr>
<tr class="even">
<td align="left">CK1</td>
<td align="right">2580</td>
</tr>
<tr class="odd">
<td align="left">PKA_Classic</td>
<td align="right">2094</td>
</tr>
<tr class="even">
<td align="left">CK2</td>
<td align="right">2073</td>
</tr>
<tr class="odd">
<td align="left">PKC_General</td>
<td align="right">1633</td>
</tr>
<tr class="even">
<td align="left">CDK_Consensus</td>
<td align="right">1327</td>
</tr>
<tr class="odd">
<td align="left">MAPK_Pxh</td>
<td align="right">1300</td>
</tr>
<tr class="even">
<td align="left">CK2_Asp</td>
<td align="right">1266</td>
</tr>
</tbody>
</table>
<p>The result <code>motif_matches</code> matches the
<code>TERM2GENE</code> format required by <code>clusterProfiler</code>:
- <code>term</code>: The motif/kinase name. - <code>gene</code>: The
site ID (Sequence + “-p”).</p>
</div>
<div class="section level2">
<h2 id="prepare-data-dpadpu">Prepare Data (DPA/DPU)<a class="anchor" aria-label="anchor" href="#prepare-data-dpadpu"></a>
</h2>
<p>We prepare the data ranking just like in standard PTM-SEA.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Prepare DPA data (Differential PTM Abundance)</span></span>
<span><span class="va">prep_dpa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ptmsea_data_prep.html">ptmsea_data_prep</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">example_data</span>,</span>
<span>  stat_column <span class="op">=</span> <span class="st">"statistic.site"</span>,</span>
<span>  seq_window_col <span class="op">=</span> <span class="st">"SequenceWindow"</span>,</span>
<span>  contrast_col <span class="op">=</span> <span class="st">"contrast"</span>,</span>
<span>  trim_to <span class="op">=</span> <span class="st">"15"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Prepare DPU data (Differential PTM Usage - protein normalized)</span></span>
<span><span class="va">prep_dpu</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ptmsea_data_prep.html">ptmsea_data_prep</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">example_data</span>,</span>
<span>  stat_column <span class="op">=</span> <span class="st">"tstatistic_I"</span>,</span>
<span>  seq_window_col <span class="op">=</span> <span class="st">"SequenceWindow"</span>,</span>
<span>  contrast_col <span class="op">=</span> <span class="st">"contrast"</span>,</span>
<span>  trim_to <span class="op">=</span> <span class="st">"15"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">prep_dpu</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "ranks"   "dropped"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="run-motif-gsea">Run Motif GSEA<a class="anchor" aria-label="anchor" href="#run-motif-gsea"></a>
</h2>
<p>We use <code>run_motif_gsea</code> to perform enrichment
analysis.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run GSEA using the scanned motif sets</span></span>
<span><span class="va">results_motif_dpa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_motif_gsea.html">run_motif_gsea</a></span><span class="op">(</span></span>
<span>  ranks_list <span class="op">=</span> <span class="va">prep_dpa</span><span class="op">$</span><span class="va">ranks</span>,</span>
<span>  motif_term2gene <span class="op">=</span> <span class="va">motif_matches</span>, <span class="co"># generated by scan_motifs</span></span>
<span>  pvalueCutoff <span class="op">=</span> <span class="fl">1.0</span>, <span class="co"># Return ALL results for visualization</span></span>
<span>  min_size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  max_size <span class="op">=</span> <span class="fl">5000</span> <span class="co"># Allow large motif sets (critical!)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">gsea_summary</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">results_motif_dpa</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://purrr.tidyverse.org/reference/map_dfr.html" class="external-link">map_dfr</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">res</span> <span class="op">&lt;-</span> <span class="va">results_motif_dpa</span><span class="op">[[</span><span class="va">name</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span></span>
<span>        Contrast <span class="op">=</span> <span class="va">name</span>,</span>
<span>        `Significant (p &lt; 0.1)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">res</span><span class="op">@</span><span class="va">result</span><span class="op">$</span><span class="va">p.adjust</span> <span class="op">&lt;</span> <span class="fl">0.1</span><span class="op">)</span></span>
<span>      <span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">gsea_summary</span>, caption <span class="op">=</span> <span class="st">"Significant Motif Sets (DPA)"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Significant Motif Sets (DPA)</caption>
<thead><tr class="header">
<th align="left">Contrast</th>
<th align="right">Significant (p &lt; 0.1)</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">KO_vs_WT</td>
<td align="right">11</td>
</tr>
<tr class="even">
<td align="left">KO_vs_WT_at_Early</td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">KO_vs_WT_at_Late</td>
<td align="right">10</td>
</tr>
<tr class="even">
<td align="left">KO_vs_WT_at_Uninfect</td>
<td align="right">9</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="visualize-results">Visualize Results<a class="anchor" aria-label="anchor" href="#visualize-results"></a>
</h2>
<div class="section level3">
<h3 id="dotplot-of-significant-motifs">Dotplot of Significant Motifs<a class="anchor" aria-label="anchor" href="#dotplot-of-significant-motifs"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define significance threshold</span></span>
<span><span class="va">sig_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">results_motif_dpa</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Filter out null results</span></span>
<span>  <span class="va">valid_res</span> <span class="op">&lt;-</span> <span class="va">results_motif_dpa</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">results_motif_dpa</span>, <span class="va">is.null</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">valid_res</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="co"># Merge results across contrasts</span></span>
<span>    <span class="va">merged_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/merge_result.html" class="external-link">merge_result</a></span><span class="op">(</span><span class="va">valid_res</span><span class="op">)</span></span>
<span></span>
<span>    <span class="co"># Check if there are any significant results</span></span>
<span>    <span class="va">has_significant</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="va">merged_res</span><span class="op">@</span><span class="va">compareClusterResult</span><span class="op">$</span><span class="va">p.adjust</span> <span class="op">&lt;</span> <span class="va">sig_threshold</span><span class="op">)</span></span>
<span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">has_significant</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="co"># Dotplot showing significant motifs</span></span>
<span>      <span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">enrichplot</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html" class="external-link">dotplot</a></span><span class="op">(</span><span class="va">merged_res</span>, showCategory <span class="op">=</span> <span class="fl">15</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Kinase Motif Enrichment (DPA)"</span>,</span>
<span>                      subtitle <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Showing motifs with p.adjust &lt;"</span>, <span class="va">sig_threshold</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="op">}</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="Analysis_Motif_Enrichment_files/figure-html/plot_dotplot-1.png" class="r-plt" width="960"></p>
</div>
<div class="section level3">
<h3 id="gsea-plot-example">GSEA Plot (Example)<a class="anchor" aria-label="anchor" href="#gsea-plot-example"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Show single example for illustration (gseaplot2 objects don't combine well)</span></span>
<span><span class="va">contrast_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">results_motif_dpa</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="va">results_motif_dpa</span><span class="op">[[</span><span class="va">contrast_name</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">&amp;&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">res</span><span class="op">@</span><span class="va">result</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">top_motif</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">@</span><span class="va">result</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">pvalue</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">nes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">top_motif</span><span class="op">$</span><span class="va">NES</span>, <span class="fl">3</span><span class="op">)</span></span>
<span>  <span class="va">padj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">signif</a></span><span class="op">(</span><span class="va">top_motif</span><span class="op">$</span><span class="va">p.adjust</span>, <span class="fl">3</span><span class="op">)</span></span>
<span></span>
<span>  <span class="fu">enrichplot</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot2.html" class="external-link">gseaplot2</a></span><span class="op">(</span></span>
<span>    <span class="va">res</span>,</span>
<span>    geneSetID <span class="op">=</span> <span class="va">top_motif</span><span class="op">$</span><span class="va">ID</span>,</span>
<span>    title <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">contrast_name</span>, <span class="st">" - "</span>, <span class="va">top_motif</span><span class="op">$</span><span class="va">ID</span>,</span>
<span>                   <span class="st">" (NES="</span>, <span class="va">nes</span>, <span class="st">", p.adj="</span>, <span class="va">padj</span>, <span class="st">")"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="Analysis_Motif_Enrichment_files/figure-html/gsea_plots_significant-1.png" class="r-plt" width="960"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.5.2 (2025-10-31)</span></span>
<span><span class="co">## Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">## Running under: macOS Tahoe 26.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib </span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Europe/Zurich</span></span>
<span><span class="co">## tzcode source: internal</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] purrr_1.2.0            fgsea_1.36.0           enrichplot_1.30.3     </span></span>
<span><span class="co">## [4] dplyr_1.1.4            clusterProfiler_4.18.1 prophosqua_0.2.0      </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] DBI_1.2.3               gson_0.1.0              rlang_1.1.6            </span></span>
<span><span class="co">##   [4] magrittr_2.0.4          DOSE_4.4.0              compiler_4.5.2         </span></span>
<span><span class="co">##   [7] RSQLite_2.4.5           png_0.1-8               systemfonts_1.3.1      </span></span>
<span><span class="co">##  [10] vctrs_0.6.5             reshape2_1.4.5          stringr_1.6.0          </span></span>
<span><span class="co">##  [13] pkgconfig_2.0.3         crayon_1.5.3            fastmap_1.2.0          </span></span>
<span><span class="co">##  [16] XVector_0.50.0          labeling_0.4.3          rmarkdown_2.30         </span></span>
<span><span class="co">##  [19] ragg_1.5.0              bit_4.6.0               xfun_0.54              </span></span>
<span><span class="co">##  [22] ggseqlogo_0.2           cachem_1.1.0            aplot_0.2.9            </span></span>
<span><span class="co">##  [25] jsonlite_2.0.0          blob_1.2.4              BiocParallel_1.44.0    </span></span>
<span><span class="co">##  [28] parallel_4.5.2          R6_2.6.1                bslib_0.9.0            </span></span>
<span><span class="co">##  [31] stringi_1.8.7           RColorBrewer_1.1-3      jquerylib_0.1.4        </span></span>
<span><span class="co">##  [34] GOSemSim_2.36.0         Rcpp_1.1.0              Seqinfo_1.0.0          </span></span>
<span><span class="co">##  [37] bookdown_0.46           knitr_1.50              ggtangle_0.0.9         </span></span>
<span><span class="co">##  [40] R.utils_2.13.0          IRanges_2.44.0          Matrix_1.7-4           </span></span>
<span><span class="co">##  [43] splines_4.5.2           igraph_2.2.1            tidyselect_1.2.1       </span></span>
<span><span class="co">##  [46] qvalue_2.42.0           yaml_2.3.11             codetools_0.2-20       </span></span>
<span><span class="co">##  [49] lattice_0.22-7          tibble_3.3.0            plyr_1.8.9             </span></span>
<span><span class="co">##  [52] withr_3.0.2             Biobase_2.70.0          treeio_1.34.0          </span></span>
<span><span class="co">##  [55] KEGGREST_1.50.0         S7_0.2.1                evaluate_1.0.5         </span></span>
<span><span class="co">##  [58] gridGraphics_0.5-1      desc_1.4.3              Biostrings_2.78.0      </span></span>
<span><span class="co">##  [61] pillar_1.11.1           ggtree_4.0.1            stats4_4.5.2           </span></span>
<span><span class="co">##  [64] ggfun_0.2.0             generics_0.1.4          rprojroot_2.1.1        </span></span>
<span><span class="co">##  [67] S4Vectors_0.48.0        ggplot2_4.0.1           scales_1.4.0           </span></span>
<span><span class="co">##  [70] tidytree_0.4.6          glue_1.8.0              gdtools_0.4.4          </span></span>
<span><span class="co">##  [73] lazyeval_0.2.2          tools_4.5.2             data.table_1.17.8      </span></span>
<span><span class="co">##  [76] ggiraph_0.9.2           fs_1.6.6                fastmatch_1.1-6        </span></span>
<span><span class="co">##  [79] cowplot_1.2.0           grid_4.5.2              tidyr_1.3.1            </span></span>
<span><span class="co">##  [82] ape_5.8-1               AnnotationDbi_1.72.0    nlme_3.1-168           </span></span>
<span><span class="co">##  [85] patchwork_1.3.2         cli_3.6.5               rappdirs_0.3.3         </span></span>
<span><span class="co">##  [88] textshaping_1.0.4       fontBitstreamVera_0.1.1 gtable_0.3.6           </span></span>
<span><span class="co">##  [91] R.methodsS3_1.8.2       yulab.utils_0.2.2       sass_0.4.10            </span></span>
<span><span class="co">##  [94] digest_0.6.39           fontquiver_0.2.1        BiocGenerics_0.56.0    </span></span>
<span><span class="co">##  [97] ggrepel_0.9.6           ggplotify_0.1.3         htmlwidgets_1.6.4      </span></span>
<span><span class="co">## [100] farver_2.1.2            memoise_2.0.1           htmltools_0.5.9        </span></span>
<span><span class="co">## [103] pkgdown_2.2.0           R.oo_1.27.1             lifecycle_1.0.4        </span></span>
<span><span class="co">## [106] here_1.0.2              httr_1.4.7              GO.db_3.22.0           </span></span>
<span><span class="co">## [109] fontLiberation_0.1.0    bit64_4.6.0-1</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Witold Wolski.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
