<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>FragPipe TMT QC • prophosqua</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><script src="QCReport_files/libs/quarto-html/popper.min.js"></script><script src="QCReport_files/libs/quarto-html/tippy.umd.min.js"></script><link href="QCReport_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="QCReport_files/libs/quarto-html/light-border.css" rel="stylesheet">
<!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="FragPipe TMT QC">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">prophosqua</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Supplementary_Material_v3.html">Integrated Analysis of Post-Translational Modifications and Total Proteome: Methods for Distinguishing Abundance from Usage Changes</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">More articles...</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/prolfqua/prophosqua/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-quarto">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>FragPipe TMT QC</h1>

      <p class="author">Functional Genomics Center Zurich</p>


      <small class="dont-index">Source: <a href="https://github.com/prolfqua/prophosqua/blob/HEAD/vignettes/QCReport.qmd" class="external-link"><code>vignettes/QCReport.qmd</code></a></small>
      <div class="d-none name"><code></code></div>
    </div>






    <section class="section level2"><h2 id="configuration">Configuration<a class="anchor" aria-label="anchor" href="#configuration"></a>
</h2>
<p>This section shows the key parameters used for this QC analysis. The report can be parameterized to analyze different datasets by providing custom file paths and identifiers.</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top">
<caption>project information</caption>
<colgroup>
<col style="width: 13%">
<col style="width: 86%">
</colgroup>
<thead><tr class="header">
<th style="text-align: left;">name</th>
<th style="text-align: left;">value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">workunit_id</td>
<td style="text-align: left;">1234</td>
</tr>
<tr class="even">
<td style="text-align: left;">project_id</td>
<td style="text-align: left;">4321</td>
</tr>
<tr class="odd">
<td style="text-align: left;">datasetname</td>
<td style="text-align: left;">ptm_example-main/qc_example_data/QCmini/psm.tsv</td>
</tr>
<tr class="even">
<td style="text-align: left;">fastasequence</td>
<td style="text-align: left;">ptm_example-main/qc_example_data/fgcz_3702_UP000006548_AraUniprot_1spg_d_20231024.fasta</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>QC will only use the first psm file.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code><span><span class="va">using</span> <span class="op">:</span> <span class="va">ptm_example</span><span class="op">-</span><span class="va">main</span><span class="op">/</span><span class="va">qc_example_data</span><span class="op">/</span><span class="va">QCmini</span><span class="op">/</span><span class="va">psm.tsv</span> </span></code></pre>
</div>
</div>
</section><section class="section level2"><h2 id="fasta-database-summary">FASTA Database Summary<a class="anchor" aria-label="anchor" href="#fasta-database-summary"></a>
</h2>
<p>The FASTA database contains the protein sequences used for peptide identification. Understanding the composition of your search database is crucial for interpreting identification results and assessing potential biases.</p>
<p><strong>Key metrics to evaluate:</strong></p>
<ul>
<li>
<strong>Database size</strong>: Typical sizes vary by organism:
<ul>
<li>Human: ~20,000 proteins</li>
<li>Mouse: ~22,000 proteins</li>
<li>Yeast: ~6,000 proteins</li>
<li>E. coli: ~4,400 proteins</li>
<li>Arabidopsis: ~27,000 protein coding genes, ~35,000 proteins</li>
</ul>
</li>
<li>
<strong>Decoy sequences</strong>: Reverse sequences (REV_) are used to estimate false discovery rates</li>
<li>
<strong>Amino acid composition</strong>: Should reflect the expected composition of your sample organism</li>
</ul>
<p>The FASTA database has 55907 sequences including decoys, and 27953 without decoys. The amino acid frequency distribution below shows the composition of your search database, which should be consistent with the expected proteome composition.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>nr sequences:
 27953
 length summary:
    Min. 1st Qu.  Median    Mean 3rd Qu.    Max.
     5.0   198.0   348.0   404.8   520.0  5400.0
 AA frequencies:
      [,1]
 A  712383
 B       5
 C  213335
 D  608123
 E  760476
 F  484702
 G  730135
 H  256377
 I  601507
 K  720098
 L 1078566
 M  277709
 N  498002
 P  543057
 Q  395258
 R  609817
 S 1031163
 T  577392
 V  754323
 W  140645
 X      15
 Y  321973
 Z       3</code></pre>
</div>
</div>
</section><section class="section level2"><h2 id="data-processing-overview">Data Processing Overview<a class="anchor" aria-label="anchor" href="#data-processing-overview"></a>
</h2>
<p>This analysis processes peptide-spectrum matches (PSMs) from FragPipe with specific quality filters to ensure reliable quantification.</p>
<p><strong>Quality filters applied:</strong></p>
<ul>
<li>
<strong>Peptide Prophet probability &gt; 0.9</strong>: Ensures high confidence identifications</li>
<li>
<strong>Abundance threshold &gt; 0</strong>: Removes PSMs without quantitative information</li>
<li>
<strong>Purity threshold = 0</strong>: No isolation purity filtering (all PSMs included)</li>
</ul>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 5436 Columns: 55
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (14): Spectrum, Spectrum File, Peptide, Modified Peptide, Prev AA, Next ...
dbl (38): Peptide Length, Charge, Retention, Observed Mass, Calibrated Obser...
lgl  (3): Observed Modifications, Is Unique, Quan Usage

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 5436 Columns: 55
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (14): Spectrum, Spectrum File, Peptide, Modified Peptide, Prev AA, Next ...
dbl (38): Peptide Length, Charge, Retention, Observed Mass, Calibrated Obser...
lgl  (3): Observed Modifications, Is Unique, Quan Usage

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<p>For this analysis we are using all PSM (Spectra) reported in the psm.tsv file with a peptide prophet probability greater than <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.9</mn><annotation encoding="application/x-tex">0.9</annotation></semantics></math>, and an abundance value in any of the channels greater then <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0</mn><annotation encoding="application/x-tex">0</annotation></semantics></math>. No other filtering is enabled. This reduces the number of PSM from 5436 to 5275.</p>
<p>The reduction in PSM count reflects the stringency of our quality filters. A typical reduction of 1-10<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>%</mi><annotation encoding="application/x-tex">\%</annotation></semantics></math> is expected and indicates proper quality control.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>INFO [2025-12-14 21:01:32] get_annot : ptm_example-main/qc_example_data/fgcz_3702_UP000006548_AraUniprot_1spg_d_20231024.fasta
INFO [2025-12-14 21:01:33] get_annot : finished reading
INFO [2025-12-14 21:01:34] get_annot : extract headers
INFO [2025-12-14 21:01:34] get_annot : all seq : 55907
INFO [2025-12-14 21:01:34] removing decoy sequences usin patter : ^REV_|^rev_
INFO [2025-12-14 21:01:34] get_annot nr seq after decoy removal: 27954
INFO [2025-12-14 21:01:34] get_annot : isUniprot : TRUE
INFO [2025-12-14 21:01:34] get_annot : extracted gene names
INFO [2025-12-14 21:01:34] get_annot : protein length
INFO [2025-12-14 21:01:35] get_annot : nr of tryptic peptides per protein computed.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in prolfquapp::dataset_protein_annot(psm, c(protein_Id = "Protein"), :
deprecated! use build_protein_annot</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>uniprot database : TRUE</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 2 pieces. Missing pieces filled with `NA` in 20 rows [54, 59, 92, 99,
178, 287, 288, 307, 357, 379, 475, 476, 554, 611, 754, 892, 893, 940, 967,
1021].</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>creating sampleName from fileName column</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in prolfqua::setup_analysis(psm, config): no isotopeLabel column
specified in the data, adding column isotopeLabel automatically and setting to
'light'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in prolfqua::setup_analysis(psm, config): no nr_children column
specified in the data, adding column nr_children and setting to 1.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>completing cases</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>completing cases done</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>setup done</code></pre>
</div>
</div>
</section><section class="section level2"><h2 id="identification-summary">Identification Summary<a class="anchor" aria-label="anchor" href="#identification-summary"></a>
</h2>
<p>These metrics provide an overview of the depth and breadth of your proteomic analysis. Higher numbers generally indicate better sample preparation and instrument performance.</p>
<p><strong>Understanding the hierarchy:</strong></p>
<ul>
<li>
<strong>Proteins</strong>: Unique protein groups identified</li>
<li>
<strong>Peptides</strong>: Unique peptide sequences (without modifications)</li>
<li>
<strong>Peptidoforms</strong>: Unique peptides with specific modifications</li>
<li>
<strong>Precursors</strong>: Unique peptide-charge state combinations</li>
<li>
<strong>PSMs</strong>: Total peptide-spectrum matches</li>
</ul>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top">
<caption>Table: Nr of proteins, peptides, peptidoforms, precursors, spectrum peptide matches overall.</caption>
<colgroup>
<col style="width: 18%">
<col style="width: 15%">
<col style="width: 15%">
<col style="width: 21%">
<col style="width: 14%">
<col style="width: 13%">
</colgroup>
<thead><tr class="header">
<th style="text-align: left;">isotopeLabel</th>
<th style="text-align: right;">protein_Id</th>
<th style="text-align: right;">peptide_Id</th>
<th style="text-align: right;">mod_peptide_Id</th>
<th style="text-align: right;">precursor</th>
<th style="text-align: right;">Spectrum</th>
</tr></thead>
<tbody><tr class="odd">
<td style="text-align: left;">light</td>
<td style="text-align: right;">1064</td>
<td style="text-align: right;">3845</td>
<td style="text-align: right;">3947</td>
<td style="text-align: right;">4860</td>
<td style="text-align: right;">5275</td>
</tr></tbody>
</table>
</div>
</div>
<p>The ratio between these levels can indicate data quality:</p>
<ul>
<li>
<strong>PSMs/Peptides ratio</strong>: Higher ratios suggest good reproducibility</li>
<li>
<strong>Peptidoforms/Peptides ratio</strong>: Indicates the extent of post-translational modifications</li>
<li>
<strong>Peptides/Proteins ratio</strong>: Reflects proteolytic efficiency and peptide diversity</li>
</ul>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure><p><img src="QCReport_files/figure-html/identSummaryPlot-1.png" width="672"></p>
<figcaption>Figure: Number of proteins, peptides, peptidoforms, ions and precursor per channel.</figcaption></figure>
</div>
</div>
</div>
<p>This plot shows the identification counts across different TMT channels. Ideally, all channels should show similar identification numbers, indicating consistent sample preparation and labeling efficiency.</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top" style="width:100%;">
<caption>Table: Nr of proteins, peptides, peptidoforms, precursors, spectrum peptide matches per channel.</caption>
<colgroup>
<col style="width: 16%">
<col style="width: 14%">
<col style="width: 13%">
<col style="width: 13%">
<col style="width: 18%">
<col style="width: 12%">
<col style="width: 11%">
</colgroup>
<thead><tr class="header">
<th style="text-align: left;">isotopeLabel</th>
<th style="text-align: left;">sampleName</th>
<th style="text-align: right;">protein_Id</th>
<th style="text-align: right;">peptide_Id</th>
<th style="text-align: right;">mod_peptide_Id</th>
<th style="text-align: right;">precursor</th>
<th style="text-align: right;">Spectrum</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">light</td>
<td style="text-align: left;">QCmini_126</td>
<td style="text-align: right;">1064</td>
<td style="text-align: right;">3845</td>
<td style="text-align: right;">3947</td>
<td style="text-align: right;">4859</td>
<td style="text-align: right;">5274</td>
</tr>
<tr class="even">
<td style="text-align: left;">light</td>
<td style="text-align: left;">QCmini_127C</td>
<td style="text-align: right;">1064</td>
<td style="text-align: right;">3845</td>
<td style="text-align: right;">3947</td>
<td style="text-align: right;">4859</td>
<td style="text-align: right;">5274</td>
</tr>
<tr class="odd">
<td style="text-align: left;">light</td>
<td style="text-align: left;">QCmini_127N</td>
<td style="text-align: right;">1064</td>
<td style="text-align: right;">3845</td>
<td style="text-align: right;">3947</td>
<td style="text-align: right;">4860</td>
<td style="text-align: right;">5274</td>
</tr>
<tr class="even">
<td style="text-align: left;">light</td>
<td style="text-align: left;">QCmini_128C</td>
<td style="text-align: right;">1064</td>
<td style="text-align: right;">3845</td>
<td style="text-align: right;">3947</td>
<td style="text-align: right;">4860</td>
<td style="text-align: right;">5275</td>
</tr>
<tr class="odd">
<td style="text-align: left;">light</td>
<td style="text-align: left;">QCmini_128N</td>
<td style="text-align: right;">1064</td>
<td style="text-align: right;">3845</td>
<td style="text-align: right;">3947</td>
<td style="text-align: right;">4859</td>
<td style="text-align: right;">5274</td>
</tr>
<tr class="even">
<td style="text-align: left;">light</td>
<td style="text-align: left;">QCmini_129C</td>
<td style="text-align: right;">1064</td>
<td style="text-align: right;">3845</td>
<td style="text-align: right;">3947</td>
<td style="text-align: right;">4859</td>
<td style="text-align: right;">5274</td>
</tr>
<tr class="odd">
<td style="text-align: left;">light</td>
<td style="text-align: left;">QCmini_129N</td>
<td style="text-align: right;">1064</td>
<td style="text-align: right;">3845</td>
<td style="text-align: right;">3947</td>
<td style="text-align: right;">4860</td>
<td style="text-align: right;">5275</td>
</tr>
<tr class="even">
<td style="text-align: left;">light</td>
<td style="text-align: left;">QCmini_130C</td>
<td style="text-align: right;">1064</td>
<td style="text-align: right;">3845</td>
<td style="text-align: right;">3947</td>
<td style="text-align: right;">4859</td>
<td style="text-align: right;">5274</td>
</tr>
<tr class="odd">
<td style="text-align: left;">light</td>
<td style="text-align: left;">QCmini_130N</td>
<td style="text-align: right;">1064</td>
<td style="text-align: right;">3845</td>
<td style="text-align: right;">3947</td>
<td style="text-align: right;">4860</td>
<td style="text-align: right;">5274</td>
</tr>
<tr class="even">
<td style="text-align: left;">light</td>
<td style="text-align: left;">QCmini_131C</td>
<td style="text-align: right;">1064</td>
<td style="text-align: right;">3844</td>
<td style="text-align: right;">3946</td>
<td style="text-align: right;">4858</td>
<td style="text-align: right;">5272</td>
</tr>
<tr class="odd">
<td style="text-align: left;">light</td>
<td style="text-align: left;">QCmini_131N</td>
<td style="text-align: right;">1064</td>
<td style="text-align: right;">3845</td>
<td style="text-align: right;">3947</td>
<td style="text-align: right;">4859</td>
<td style="text-align: right;">5273</td>
</tr>
<tr class="even">
<td style="text-align: left;">light</td>
<td style="text-align: left;">QCmini_132C</td>
<td style="text-align: right;">1064</td>
<td style="text-align: right;">3845</td>
<td style="text-align: right;">3947</td>
<td style="text-align: right;">4859</td>
<td style="text-align: right;">5273</td>
</tr>
<tr class="odd">
<td style="text-align: left;">light</td>
<td style="text-align: left;">QCmini_132N</td>
<td style="text-align: right;">1064</td>
<td style="text-align: right;">3845</td>
<td style="text-align: right;">3947</td>
<td style="text-align: right;">4860</td>
<td style="text-align: right;">5274</td>
</tr>
<tr class="even">
<td style="text-align: left;">light</td>
<td style="text-align: left;">QCmini_133C</td>
<td style="text-align: right;">1064</td>
<td style="text-align: right;">3845</td>
<td style="text-align: right;">3947</td>
<td style="text-align: right;">4859</td>
<td style="text-align: right;">5274</td>
</tr>
<tr class="odd">
<td style="text-align: left;">light</td>
<td style="text-align: left;">QCmini_133N</td>
<td style="text-align: right;">1064</td>
<td style="text-align: right;">3845</td>
<td style="text-align: right;">3947</td>
<td style="text-align: right;">4859</td>
<td style="text-align: right;">5274</td>
</tr>
<tr class="even">
<td style="text-align: left;">light</td>
<td style="text-align: left;">QCmini_134C</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">light</td>
<td style="text-align: left;">QCmini_134N</td>
<td style="text-align: right;">1064</td>
<td style="text-align: right;">3845</td>
<td style="text-align: right;">3947</td>
<td style="text-align: right;">4859</td>
<td style="text-align: right;">5274</td>
</tr>
<tr class="even">
<td style="text-align: left;">light</td>
<td style="text-align: left;">QCmini_135N</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">9</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Quality indicators:</strong></p>
<ul>
<li>
<strong>Consistent counts across channels</strong>: Good sample preparation</li>
<li>
<strong>Large variations</strong>: May indicate labeling issues or sample degradation</li>
<li>
<strong>Outlier channels</strong>: Should be investigated for technical problems</li>
</ul></section><section class="section level2"><h2 id="modifications-summary">Modifications Summary<a class="anchor" aria-label="anchor" href="#modifications-summary"></a>
</h2>
<p>Post-translational modifications (PTMs) and chemical modifications provide insights into sample treatment and biological state. TMT labeling introduces specific modifications that should be monitored for labeling efficiency.</p>
<p><strong>Key modifications to monitor:</strong></p>
<ul>
<li>
<strong>TMT labels</strong>: N-term(229.1629), K(229.1629) for TMT10plex</li>
<li>
<strong>Oxidation</strong>: M(15.9949) - common artifact</li>
<li>
<strong>Carbamylation</strong>: N-term/K modifications from urea treatment</li>
</ul>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top">
<caption>Number and type of modifications observed in the data.</caption>
<thead><tr class="header">
<th style="text-align: left;">Var1</th>
<th style="text-align: right;">Freq</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">C(57.0214)</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">K(304.2071)</td>
<td style="text-align: right;">2339</td>
</tr>
<tr class="odd">
<td style="text-align: left;">M(15.9949)</td>
<td style="text-align: right;">635</td>
</tr>
<tr class="even">
<td style="text-align: left;">N-term(304.2071)</td>
<td style="text-align: right;">3904</td>
</tr>
<tr class="odd">
<td style="text-align: left;">N-term(42.0106)</td>
<td style="text-align: right;">10</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure><p><img src="QCReport_files/figure-html/modSummaryPlot-1.png" width="672"></p>
<figcaption>Number and type of modification observed in the data.</figcaption></figure>
</div>
</div>
</div>
<p>The modification landscape reflects both intentional chemical treatments (TMT labeling) and unintentional modifications (oxidation, deamidation). High numbers of TMT modifications indicate successful labeling.</p>
</section><section class="section level2"><h2 id="labelling-efficiency">Labelling Efficiency<a class="anchor" aria-label="anchor" href="#labelling-efficiency"></a>
</h2>
<p>TMT labeling efficiency is critical for accurate quantification. Incomplete labeling leads to quantitative errors and reduced dynamic range.</p>
<p><strong>Monitoring labeling efficiency:</strong></p>
<ul>
<li>
<strong>N-terminal labeling</strong>: Should approach 100% for most peptides</li>
<li>
<strong>Lysine labeling</strong>: Should be &gt;95% for high-quality samples</li>
<li>
<strong>Unlabeled peptides</strong>: May co-elute and interfere with quantification</li>
</ul>
<section class="level2"><h3 id="n-term">N-term<a class="anchor" aria-label="anchor" href="#n-term"></a>
</h3>
<p>N-terminal labeling efficiency is typically very high (&gt;95%) as it’s kinetically favored.</p>
<section class="level3"><h4 id="peptides">Peptides<a class="anchor" aria-label="anchor" href="#peptides"></a>
</h4>
<ul>
<li>Total number of peptides : 3947</li>
<li>Number of peptides with modified N-term : 3904</li>
<li>Percent peptides with modified N-term: 99 %</li>
</ul>
<p><strong>Expected range</strong>: 95-100%. Values below 95% may indicate:</p>
<ul>
<li>Suboptimal labeling conditions</li>
<li>Sample degradation</li>
<li>Buffer incompatibility</li>
</ul></section><section class="level3"><h4 id="psms">PSM’s<a class="anchor" aria-label="anchor" href="#psms"></a>
</h4>
<ul>
<li>Total number of PSMs : 5275</li>
<li>Number of PSMs with modified N-term : 5227</li>
<li>Percent PSMs with modified N-term: 99 %</li>
</ul>
<p>PSM-level statistics weight peptides by their identification frequency, providing insight into the most abundant species in your sample.</p>
</section></section><section class="level2"><h3 id="lysine-modification">Lysine Modification<a class="anchor" aria-label="anchor" href="#lysine-modification"></a>
</h3>
<p>Lysine labeling is more challenging than N-terminal labeling and is more sensitive to reaction conditions.</p>
<section class="level3"><h4 id="peptides-1">Peptides<a class="anchor" aria-label="anchor" href="#peptides-1"></a>
</h4>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Use of .data in tidyselect expressions was deprecated in tidyselect 1.2.0.
ℹ Please use `"modSeq"` instead of `.data$modSeq`</code></pre>
</div>
</div>
<ul>
<li>Total number of peptides with Lysine: 2339</li>
<li>Number of peptides with modified Lysine residues : 2339</li>
<li>Percent peptides with modified Lysine residues: 100 %</li>
</ul>
<p><strong>Expected range</strong>: 95-99%. Lower values may indicate:</p>
<ul>
<li>Insufficient reagent concentration</li>
<li>Competing reactions (e.g., formaldehyde crosslinking)</li>
<li>pH optimization needed</li>
</ul></section><section class="level3"><h4 id="psms-with-lysine">PSM’s with Lysine<a class="anchor" aria-label="anchor" href="#psms-with-lysine"></a>
</h4>
<ul>
<li>Total number of PSMs with Lysine: 3200</li>
<li>Number of PSMs with modified Lysine residues : 3200</li>
<li>Percent PSMs with modified Lysine residues: 100 %</li>
</ul>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Use of .data in tidyselect expressions was deprecated in tidyselect 1.2.0.
ℹ Please use `"n"` instead of `.data$n`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Use of .data in tidyselect expressions was deprecated in tidyselect 1.2.0.
ℹ Please use `"n1"` instead of `.data$n1`</code></pre>
</div>
</div>
</section><section class="level3"><h4 id="peptidoforms">Peptidoforms<a class="anchor" aria-label="anchor" href="#peptidoforms"></a>
</h4>
<ul>
<li>Total number of Lysine residues: 2344</li>
<li>Number of modified Lysine residues : 2339</li>
<li>Percent modified Lysine residues: 100 %</li>
</ul></section><section class="level3"><h4 id="total-number-of-lysine-residues-in-psms">Total number of Lysine residues in PSM’s<a class="anchor" aria-label="anchor" href="#total-number-of-lysine-residues-in-psms"></a>
</h4>
<ul>
<li>Total number of Lysine residues: 3205</li>
<li>Number of modified Lysine residues : 3200</li>
<li>Percent modified Lysine residues when taking number of PSMs into account: 100 %</li>
</ul></section></section></section><section class="section level2"><h2 id="quantitative-information-per-channel">Quantitative information per channel<a class="anchor" aria-label="anchor" href="#quantitative-information-per-channel"></a>
</h2>
<p>TMT quantification relies on consistent labeling and equal loading across channels. These plots help identify technical issues that could bias quantitative comparisons.</p>
<p><strong>Quality indicators:</strong></p>
<ul>
<li>
<strong>Similar total abundances</strong>: Good sample preparation and loading</li>
<li>
<strong>Outlier channels</strong>: May indicate pipetting errors or sample loss</li>
<li>
<strong>Systematic patterns</strong>: Could suggest batch effects or labeling issues</li>
</ul>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure><p><img src="QCReport_files/figure-html/totalQuant-1.png" width="672"></p>
<figcaption>Total abundance per channel (Sum of all abundances).</figcaption></figure>
</div>
</div>
</div>
<p>Total abundance reflects both sample amount and ionization efficiency. Large variations (&gt;2-fold) should be investigated.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure><p><img src="QCReport_files/figure-html/relativeQuant-1.png" width="672"></p>
<figcaption>Ralative to chanell 126 total abundance per chanel.</figcaption></figure>
</div>
</div>
</div>
<p>Normalization to a reference channel (126) helps identify systematic biases. Values should typically be within 0.5-2.0 fold of the reference.</p>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `aes_string()` was deprecated in ggplot2 3.0.0.
ℹ Please use tidy evaluation idioms with `aes()`.
ℹ See also `vignette("ggplot2-in-packages")` for more information.
ℹ The deprecated feature was likely used in the prolfqua package.
  Please report the issue at &lt;https://github.com/wolski/prolfqua/issues&gt;.</code></pre>
</div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure><p><img src="QCReport_files/figure-html/densityQuant-1.png" width="672"></p>
<figcaption>Density of abundance values per channel.</figcaption></figure>
</div>
</div>
</div>
<p>The abundance distribution should be similar across channels. Shifted distributions may indicate: - Unequal sample loading - Different sample complexity - Technical artifacts in specific channels</p>
</section><section class="section level2"><h2 id="missed-cleavage">Missed cleavage<a class="anchor" aria-label="anchor" href="#missed-cleavage"></a>
</h2>
<p>Missed cleavage analysis provides insights into proteolytic efficiency and sample quality. Excessive missed cleavages can reduce identification rates and affect quantification accuracy.</p>
<p>Missed cleavage site: Is a residue after which trypsin should have cleaved but did not.</p>
<p><strong>Factors affecting missed cleavages:</strong></p>
<ul>
<li>
<strong>Protein denaturation</strong>: Incomplete unfolding reduces accessibility</li>
<li>
<strong>Digestion time/temperature</strong>: Insufficient conditions lead to incomplete digestion</li>
<li>
<strong>Enzyme activity</strong>: Old or inhibited trypsin reduces efficiency</li>
<li>
<strong>Chemical modifications</strong>: Modified residues may resist cleavage</li>
</ul>
<p>To determine the number of missed cleavages we compute: - The number of all potential cleavage sites (i.e., number of K or R residues) - The number of actual cleavage sites (K or R at peptide C-terminus) - Modified residues that may resist cleavage</p>
<section class="level2"><h3 id="missed-lysine-residues">Missed Lysine residues<a class="anchor" aria-label="anchor" href="#missed-lysine-residues"></a>
</h3>
<p>We compute the total number of K residues, and the number of K cleavage sites (nr of potential cleavage sites). Then we compute the number of K at the C-term</p>
<ul>
<li>The number of K residues 3205</li>
<li>The number of unmodified K at C term 0</li>
<li>The number of modified K at C term 3025</li>
<li>The number of any K at C term 3025</li>
<li>The number of any modified K: 3200</li>
<li>Missed cleavage sites : number of K residues - number of any K at C term = 180,</li>
<li>and in % of number of K residues : 6</li>
</ul>
<p><strong>Expected range</strong>: 5-15% missed cleavages are typical. Higher rates may indicate:</p>
<ul>
<li>Incomplete protein denaturation</li>
<li>Insufficient digestion time</li>
<li>Trypsin inhibitors present</li>
<li>High protein concentration</li>
</ul></section><section class="level2"><h3 id="missed-arginine-residues">Missed Arginine residues<a class="anchor" aria-label="anchor" href="#missed-arginine-residues"></a>
</h3>
<ul>
<li>The number of R residues 2401</li>
<li>The number of unmodified R at C term 2189</li>
<li>The number of modified R at C term 0</li>
<li>The number of any R at C term 2189</li>
<li>The number of any modified R: 0</li>
<li>Missed cleavage sites : number of R residues - number of any R at C term = 212,</li>
<li>and in % of number of R residues : 9</li>
</ul>
<p>Arginine cleavage is generally more efficient than lysine cleavage. Similar missed cleavage rates between K and R suggest consistent digestion conditions.</p>
</section></section><section class="section level2"><h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="cell">
<div class="cell-output-display">
<p><strong>R version 4.5.2 (2025-10-31)</strong></p>
<p><strong>Platform:</strong> aarch64-apple-darwin20</p>
<p><strong>locale:</strong> en_US.UTF-8||en_US.UTF-8||en_US.UTF-8||C||en_US.UTF-8||en_US.UTF-8</p>
<p><strong>attached base packages:</strong> <em>stats</em>, <em>graphics</em>, <em>grDevices</em>, <em>utils</em>, <em>datasets</em>, <em>methods</em> and <em>base</em></p>
<p><strong>other attached packages:</strong> <em>here(v.1.0.2)</em>, <em>ggseqlogo(v.0.2)</em>, <em>lubridate(v.1.9.4)</em>, <em>forcats(v.1.0.1)</em>, <em>stringr(v.1.6.0)</em>, <em>dplyr(v.1.1.4)</em>, <em>purrr(v.1.2.0)</em>, <em>readr(v.2.1.6)</em>, <em>tidyr(v.1.3.1)</em>, <em>tibble(v.3.3.0)</em>, <em>ggplot2(v.4.0.1)</em>, <em>tidyverse(v.2.0.0)</em>, <em>prophosqua(v.0.1.0)</em> and <em>prolfquapp(v.0.1.8)</em></p>
<p><strong>loaded via a namespace (and not attached):</strong> <em>AhoCorasickTrie(v.0.1.3)</em>, <em>gtable(v.0.3.6)</em>, <em>xfun(v.0.54)</em>, <em>prozor(v.0.3.1)</em>, <em>htmlwidgets(v.1.6.4)</em>, <em>ggrepel(v.0.9.6)</em>, <em>lattice(v.0.22-7)</em>, <em>tzdb(v.0.5.0)</em>, <em>vctrs(v.0.6.5)</em>, <em>tools(v.4.5.2)</em>, <em>generics(v.0.1.4)</em>, <em>parallel(v.4.5.2)</em>, <em>pkgconfig(v.2.0.3)</em>, <em>pheatmap(v.1.0.13)</em>, <em>Matrix(v.1.7-4)</em>, <em>data.table(v.1.17.8)</em>, <em>RColorBrewer(v.1.1-3)</em>, <em>S7(v.0.2.1)</em>, <em>lifecycle(v.1.0.4)</em>, <em>prolfqua(v.1.4.0)</em>, <em>compiler(v.4.5.2)</em>, <em>farver(v.2.1.2)</em>, <em>codetools(v.0.2-20)</em>, <em>htmltools(v.0.5.9)</em>, <em>lazyeval(v.0.2.2)</em>, <em>yaml(v.2.3.11)</em>, <em>plotly(v.4.11.0)</em>, <em>pillar(v.1.11.1)</em>, <em>crayon(v.1.5.3)</em>, <em>seqinr(v.4.2-36)</em>, <em>MASS(v.7.3-65)</em>, <em>cachem(v.1.1.0)</em>, <em>tidyselect(v.1.2.1)</em>, <em>conflicted(v.1.2.0)</em>, <em>digest(v.0.6.39)</em>, <em>stringi(v.1.8.7)</em>, <em>pander(v.0.6.6)</em>, <em>labeling(v.0.4.3)</em>, <em>ade4(v.1.7-23)</em>, <em>rprojroot(v.2.1.1)</em>, <em>fastmap(v.1.2.0)</em>, <em>grid(v.4.5.2)</em>, <em>cli(v.3.6.5)</em>, <em>logger(v.0.4.1)</em>, <em>magrittr(v.2.0.4)</em>, <em>patchwork(v.1.3.2)</em>, <em>withr(v.3.0.2)</em>, <em>scales(v.1.4.0)</em>, <em>bit64(v.4.6.0-1)</em>, <em>timechange(v.0.3.0)</em>, <em>httr(v.1.4.7)</em>, <em>rmarkdown(v.2.30)</em>, <em>bit(v.4.6.0)</em>, <em>gridExtra(v.2.3)</em>, <em>hms(v.1.1.4)</em>, <em>memoise(v.2.0.1)</em>, <em>evaluate(v.1.0.5)</em>, <em>knitr(v.1.50)</em>, <em>viridisLite(v.0.4.2)</em>, <em>dtplyr(v.1.3.2)</em>, <em>rlang(v.1.1.6)</em>, <em>Rcpp(v.1.1.0)</em>, <em>docopt(v.0.7.2)</em>, <em>glue(v.1.8.0)</em>, <em>vroom(v.1.6.7)</em>, <em>jsonlite(v.2.0.0)</em> and <em>R6(v.2.6.1)</em></p>
</div>
</div>
</section><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'light-border',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config);
    }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script></main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jonas Grossmann, Witold Wolski.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
